dags/hindawi/parser.py

```diff
@@ -0,0 +1,201 @@
+                author_["orcid"] = orcid.text
+            authors.append(author_)
+
+        return authors
+
+    def _get_affiliations(self, author):
+        affiliations = [
+            affiliation.text.replace("\n", "")
+            for affiliation in author.findall("ns0:subfield[@code='u']", self.prefixes)
+        ]
+        parsed_affiliations = [
+            {
+                "value": affiliation,
+                "organization": (",").join(affiliation.split(",")[:-1]),
+                "country": re.findall(r"[A-Z][a-z]*$", affiliation)[0],
+            }
+            for affiliation in affiliations
+        ]
+        return parsed_affiliations
+
+    def _get_arxiv(self, article: ET.Element):
+        arxiv = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='9']",
+            self.prefixes,
+        )
+        if arxiv is None:
+            self.logger.error("No arxiv id_get_copyright_statement found.")
+            return None
+        if arxiv.text.lower() == "arxiv":
+            arxiv_value = article.find(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='a']",
+                self.prefixes,
+            ).text.lower()
+            return [{"value": arxiv_value.replace("arxiv:", "").strip()}]
+
+    def _get_journal_pages(self, article: ET.Element):
+        journal_pages = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='773']/ns0:subfield/[@code='c']",
+            self.prefixes,
+        ).text
+        if "-" in journal_pages:
+            return journal_pages.split("-", 1)
+        else:
+            return journal_pages, ""
+
+    def _construct_license(self, url, license_type, version):
+        if url and license_type and version:
+            return {
+                "url": url,
+                "license": f"CC-{license_type}-{version}",
+            }
+
+    def _get_license(self, article: ET.Element):
+        licenses = []
+        license_urls = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='540']/ns0:subfield/[@code='u']",
+            self.prefixes,
+        )
+        license_texts = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='540']/ns0:subfield/[@code='a']",
+            self.prefixes,
+        )
+        for license_url, license_text in zip(license_urls, license_texts):
+            if license_url.text:
+                url_parts = license_url.text.split("/")
+                clean_url_parts = list(filter(bool, url_parts))
+                version = clean_url_parts.pop()
+                licencse_type = clean_url_parts.pop()
+                licenses.append(
+                    self._construct_license(
+                        license_url.text, licencse_type.upper(), version
+                    )
+                )
+            elif license_text.text:
+                licenses.append(license_text.text)
+            return licenses
```

Comment:
don't we have another publisher with the same logic?

Reply:
also, you are returning inside the `for` loop. Is it what we want?

Reply:
similar, but not the same. Springer looks really similar, but actually, in hindawi, we're getting url and licenses statement, while in Springer we are getting license type and version, no statement

Reply:
and no, we don't want to return inside of for loop